source/upgrade/kolla.rst

```diff
@@ -5,210 +5,272 @@ Kolla
+HAProxy
+-------
 
-* Horizon: for versions < Rocky: when using an overlay configuration file sync it with the new version from https://raw.githubusercontent.com/osism/cfg-cookiecutter/master/cfg-%7B%7Bcookiecutter.project_name%7D%7D/environments/kolla/files/overlays/horizon/local_settings.j2.RELEASE
+* When updating to *Rocky* - Synchronize `haproxy.cfg`_ to
+  ``environments/kolla/files/overlays/haproxy/haproxy.cfg``.
 
-  * if you upgrade to >= Rocky, this file is removed from the repository
+* When updating to *Stein* - Synchronize `haproxy_main.cfg`_ to
+  ``environments/kolla/files/overlays/haproxy/haproxy_main.cfg``. The old
+  ``environments/kolla/files/overlays/haproxy/haproxy.cfg`` need to be deleted.
 
-* Horizon: for versions >= Rocky: when using an overlay configuration file environments/kolla/files/overlays/horizon/local_settings.j2 , replace it by https://raw.githubusercontent.com/osism/cfg-cookiecutter/master/cfg-%7B%7Bcookiecutter.project_name%7D%7D/environments/kolla/files/overlays/horizon/custom_local_settings
+.. _haproxy.cfg: https://raw.githubusercontent.com/osism/cfg-cookiecutter/master/cfg-%7B%7Bcookiecutter.project_name%7D%7D/environments/kolla/files/overlays/haproxy/haproxy.cfg.rocky
+.. _haproxy_main.cfg: https://raw.githubusercontent.com/osism/cfg-cookiecutter/master/cfg-%7B%7Bcookiecutter.project_name%7D%7D/environments/kolla/files/overlays/haproxy/haproxy_main.cfg.stein
 
-* gather facts with ``osism-generic facts`` before the upgrade
+Horizon
+-------
 
-* backup MariaDB databases before the upgrade
+* The file ``environments/kolla/files/overlays/horizon/local_settings.j2`` need
+  to be removed.
 
-Notes
-=====
+* Copy `custom_local_settings`_ to
+  ``environments/kolla/files/overlays/horizon/custom_local_settings``.
 
-* Horizon: after the upgrade cleanup and regenerate the cached files with ``docker exec -it horizon rm /var/lib/kolla/.local_settings.md5sum.txt && docker restart horizon``
+.. _custom_local_settings: https://raw.githubusercontent.com/osism/cfg-cookiecutter/master/cfg-%7B%7Bcookiecutter.project_name%7D%7D/environments/kolla/files/overlays/horizon/custom_local_settings
 
-* Nova: Upgrade the controller (``osism-kolla upgrade nova -l controller``) followed by the compute nodes (``osism-kolla upgrade nova -l compute``)
+* Gather facts with before the upgrade
 
-* Elasticsearch: After the ugprade of Elasticsearch enable the shard allocation.
+.. code-block:: console
 
-  .. code-block:: console
+   osism-generic facts
 
-     curl -X PUT "http://api-int.osism.local:9200/_cluster/settings?pretty" -H 'Content-Type: application/json' -d'
-     {
-       "persistent": {
-         "cluster.routing.allocation.enable": null
-       }
-     }
-     '
+* Backup MariaDB databases before the upgrade on one of the control nodes:
```

Comment:
Use a reference to the MariaDB backup section here (http://docs.osism.io/operations/mariadb.html#backup).